src/worker.py

```diff
@@ -162,10 +163,16 @@ def __init__(self, directory, hidden=False):
 
     def start(self):
         self.emit("started", self._directory)
-        self._paths = os.listdir(self._directory)
-        self._total = len(self._paths)
-        self._index = 0
-        GLib.idle_add(self.step, priority=GLib.PRIORITY_HIGH_IDLE + 20)
+        try:
```

Comment:
I suspect the try is only needed for `os.listdir`.

Reply:
Yes, it's only required for the `listdir`, I let the full block to prevent a award looking `return`, but I make the change!